torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -172,8 +171,10 @@ void removeExtraLoopOutputs(Node* node) {
   for (size_t i_1 = node->outputs().size(); i_1 > 0; --i_1) {
     size_t i = i_1 - 1;
     // if the value is no longer changed remove output
-    if (loop_body->inputs().at(loop_body_offset + i) ==
-        loop_body->outputs().at(loop_body_offset + i)) {
+    if ((loop_body->inputs().at(loop_body_offset + i) ==
+         loop_body->outputs().at(loop_body_offset + i)) ||
+        (node->inputs().at(loop_input_offset + i)) ==
```

Comment:
pleeeeese add a comment. This change is non obvious.

Reply:
I think applying the same here fix to DCE would have also fixed the jitter - haven't checked yet though

Reply:
Yeah, this place is pretty hard to understand. Actually, it's not obvious to me how it worked even before your changes - i.e. I don't understand where these magic numbers `loop_input_offset` and  `loop_body_offset` come from. If you understand it better, could you please add a comment how this function works?

Reply:
it's the loop node format. 

  // the format of the Loop instruction is:
  // loop_carried_outputs* = Loop(max_trip_count, start_condition,
  //                              loop_carried_inputs*)
  //                    block0(loop_counter, loop_carried_block*) {
  //                       <body>
  //                       -> (continue_condition, loop_carried_block_outputs*)
  //                    }

Reply:
Thanks for the explanation and updating the comment! This part also looks good to me now!

(Optional) nit: maybe add `const` to `loop_input_offset` and `loop_body_offset`.